File: qe-selenium-java/Activity7.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity7 {
public static void main(String[] args) {
    WebDriver driver = new FirefoxDriver();
    driver.get("https://training-support.net/webelements/dynamic-controls");
    System.out.println(driver.getTitle());

    
    System.out.println("Checking if text Input Field is Enabled or not!");
    WebElement textField = driver.findElement(By.xpath("//input[@id = 'textInput']"));
    System.out.println("Is textField enabled? "+ textField.isEnabled());

    

    driver.findElement(By.xpath("//button[text() = 'Enable Input']")).click();
    System.out.println("Clicked Enable Input Button!");
    System.out.println("Is textField enabled? " + textField.isEnabled());

    driver.quit();
    
}
}

File: qe-selenium-python/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By


# when we have with 

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-controls")
    print(driver.title)

    text_input = driver.find_element(By.XPATH, "//input[@id = 'textInput']")
    print(text_input.is_enabled())

    driver.find_element(By.XPATH, "//button[text() = 'Enable Input']").click()
    print(text_input.is_enabled())

    driver.quit()
    


